src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1023,10 +1024,11 @@ void CompileBroker::init_compiler_sweeper_threads() {
     if (!UseDynamicNumberOfCompilerThreads || i == 0) {
       JavaThread *ct = make_thread(compiler_t, thread_handle, _c1_compile_queue, _compilers[0], THREAD);
       assert(ct != NULL, "should have been handled for initial thread");
+      ThreadsListHandle tlh;  // get_thread_name() depends on the TLH.
+      assert(tlh.includes(ct), "ct=" INTPTR_FORMAT " exited unexpectedly.", p2i(ct));
```

Comment:
This can be placed inside the if-clause where the MutexLocker was.

Reply:
True. That would be better.